api/baremetal/v1/server_utils.go

```diff
@@ -207,3 +207,58 @@ func (s *API) WaitForServerOptions(req *WaitForServerOptionsRequest, opts ...scw
 
 	return server.(*Server), nil
 }
+
+// WaitForServerPrivateNetworksRequest is used by WaitForServerPrivateNetworks method.
+type WaitForServerPrivateNetworksRequest struct {
+	ServerID      string
+	Zone          scw.Zone
+	Timeout       *time.Duration
+	RetryInterval *time.Duration
+}
+
+// WaitForServerPrivateNetworks wait for all server private networks to be in a "terminal state" before returning.
+// This function can be used to wait for all server private networks to be set.
+func (s *PrivateNetworkAPI) WaitForServerPrivateNetworks(req *WaitForServerPrivateNetworksRequest, opts ...scw.RequestOption) (*ServerPrivateNetwork, error) {
+	timeout := defaultTimeout
+	if req.Timeout != nil {
+		timeout = *req.Timeout
+	}
+	retryInterval := defaultRetryInterval
+	if req.RetryInterval != nil {
+		retryInterval = *req.RetryInterval
+	}
+
+	terminalStatus := map[ServerPrivateNetworkStatus]struct{}{
+		ServerPrivateNetworkStatusAttached: {},
+		ServerPrivateNetworkStatusError:    {},
+		ServerPrivateNetworkStatusUnknown:  {},
```

Comment:
Is this a terminal state? can you confirm that with the team?

Reply:
Yes it is